src/NewTwitchApi/Cli/CliClient.php

```diff
@@ -0,0 +1,102 @@
+use NewTwitchApi\Cli\CliEndpoints\GetAppAccessTokenCliEndpoint;
+use NewTwitchApi\Cli\CliEndpoints\GetGamesCliEndpoint;
+use NewTwitchApi\Cli\CliEndpoints\GetStreamsCliEndpoint;
+use NewTwitchApi\Cli\CliEndpoints\GetUsersCliEndpoint;
+use NewTwitchApi\Cli\CliEndpoints\GetUsersFollowsCliEndpoint;
+use NewTwitchApi\Cli\CliEndpoints\GetUserWithAccessTokenCliEndpoint;
+use NewTwitchApi\Cli\CliEndpoints\GetWebhookSubscriptionsCliEndpoint;
+use NewTwitchApi\Cli\CliEndpoints\RefreshTokenCliEndpoint;
+use NewTwitchApi\Cli\CliEndpoints\ValidateTokenCliEndpoint;
+use NewTwitchApi\Cli\Exceptions\ExitCliException;
+use NewTwitchApi\Cli\IO\InputOutput;
+use NewTwitchApi\Cli\IO\InputReader;
+use NewTwitchApi\Cli\IO\OutputWriter;
+use NewTwitchApi\Cli\IO\Stdin;
+use NewTwitchApi\HelixGuzzleClient;
+use NewTwitchApi\NewTwitchApi;
+
+class CliClient
+{
+    /** @var array */
+    private $endpoints;
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $argv)
+    {
+        if (!isset($argv[1], $argv[2])) {
+            throw new InvalidArgumentException(sprintf('Usage: php %s <client-id> <client-secret>', $argv[0]));
+        }
+
+        $clientId = $argv[1];
+        $clientSecret = $argv[2];
+
+        $helixGuzzleClient = new HelixGuzzleClient($clientId);
+        $newTwitchApi = new NewTwitchApi($helixGuzzleClient, $clientId, $clientSecret);
+        $inputOutput = new InputOutput(
+            new InputReader(new Stdin()),
+            new OutputWriter()
+        );
+        $this->endpoints = [
+            new ExitCliEndpoint(),
+            new ValidateTokenCliEndpoint($newTwitchApi, $inputOutput),
+            new RefreshTokenCliEndpoint($newTwitchApi, $inputOutput),
+            new GetAppAccessTokenCliEndpoint($newTwitchApi, $inputOutput),
+            new GetGamesCliEndpoint($newTwitchApi, $inputOutput),
+            new GetStreamsCliEndpoint($newTwitchApi, $inputOutput),
+            new GetUsersCliEndpoint($newTwitchApi, $inputOutput),
+            new GetUserWithAccessTokenCliEndpoint($newTwitchApi, $inputOutput),
+            new GetUsersFollowsCliEndpoint($newTwitchApi, $inputOutput),
+            new GetWebhookSubscriptionsCliEndpoint($newTwitchApi, $inputOutput),
+        ];
+    }
+
+    public function run(): void
+    {
+        echo 'Twitch API Testing Tool'.PHP_EOL;
+
+        while (true) {
+            try {
+                $endpoint = $this->promptForEndpoint();
+                echo $endpoint->getName() . PHP_EOL;
+                $requestResponse = $endpoint->execute();
+                echo PHP_EOL . $requestResponse->getRequest()->getRequestTarget() . PHP_EOL;
+                echo PHP_EOL . json_encode(json_decode($requestResponse->getResponse()->getBody()->getContents()), JSON_PRETTY_PRINT) . PHP_EOL;
+            } catch (ExitCliException $e) {
+                exit;
+            } catch (Exception $e) {
+                echo $e->getMessage().PHP_EOL;
```

Comment:
Consider adding a non-zero exit code here with `exit(-1)`;

Reply:
I don't want to exit here at all. You can see in the README section for the CLI client (or if you run it yourself) that this Exception is from Guzzle, typically meaning we got a 4XX code. I don't want to exit there. I put the user back at the selection menu to try again or try something different. The only actual exit condition is choosing to exit at the menu (aside from Ctrl-C, of course), which is handled by the `ExitCliException` above.

Reply:
Silly me, missed that!